commit 0970d06ae27d5558894ea07cdd168672a9fbb58a
Author: Andreas Schwab <schwab@linux-m68k.org>
Date:   Wed Aug 29 15:41:51 2012 +0200

    Fix more uses of EXT2_IOC_[GS]ETFLAGS

Index: libarchive-3.0.4/libarchive/archive_read_disk_posix.c
===================================================================
--- libarchive-3.0.4.orig/libarchive/archive_read_disk_posix.c	2013-04-21 00:21:30.502368500 +0000
+++ libarchive-3.0.4/libarchive/archive_read_disk_posix.c	2013-04-21 00:21:30.418368497 +0000
@@ -984,7 +984,7 @@
 #elif defined(EXT2_IOC_GETFLAGS) && defined(EXT2_NODUMP_FL) &&\
       defined(HAVE_WORKING_EXT2_IOC_GETFLAGS)
 		if (S_ISREG(st->st_mode) || S_ISDIR(st->st_mode)) {
-			unsigned long stflags;
+			int stflags;
 
 			t->entry_fd = open_on_current_dir(t,
 			    tree_current_access_path(t), O_RDONLY | O_NONBLOCK);
Index: libarchive-3.0.4/libarchive/archive_write_disk_posix.c
===================================================================
--- libarchive-3.0.4.orig/libarchive/archive_write_disk_posix.c	2013-04-21 00:21:30.502368500 +0000
+++ libarchive-3.0.4/libarchive/archive_write_disk_posix.c	2013-04-21 00:21:30.422368498 +0000
@@ -2403,8 +2403,8 @@
 {
 	int		 ret;
 	int		 myfd = fd;
-	unsigned long newflags, oldflags;
-	unsigned long sf_mask = 0;
+	int newflags, oldflags;
+	int sf_mask = 0;
 
 	if (set == 0  && clear == 0)
 		return (ARCHIVE_OK);
